nixos/modules/services/web-servers/keter/default.nix

```diff
@@ -0,0 +1,163 @@
+{ config, pkgs, lib, ... }:
+let
+  cfg = config.services.keter;
+in
+{
+  meta = {
+    maintainers = with lib.maintainers; [ jappie ];
+  };
+
+  options.services.keter = {
+    enable = lib.mkEnableOption ''keter, a web app deployment manager.
+        Note that this module only support loading of webapps:
+        Keep an old app running and swap the ports when the new one is booted.
+    '';
+
+    keterRoot = lib.mkOption {
+      type = lib.types.str;
+      default = "/var/lib/keter";
+      description = ''Mutable state folder for keter'';
+    };
+
+    keterPackage = lib.mkOption {
+      type = lib.types.package;
+      default = null;
+      description = ''The keter package to be used, usually <varname>pkgs.haskellPackages.keter</varname>'';
+    };
+
+    globalKeterConfig = lib.mkOption {
+      type = lib.types.attrs;
+      default = {
+        ip-from-header = true;
+        listeners = [{
+          host = "*4";
+          port = 6981;
+        }];
+      };
+      # You want that ip-from-header in the nginx setup case
+      # so it's not set to 127.0.0.1.
+      # using a port above 1024 allows you to avoid needing CAP_NET_BIND_SERVICE
+      defaultText = lib.literalExpression ''
+        {ip-from-header = true; listeners = [{host = "*4"; port = 6981;}];}
```

Comment:
Format our little nixexpr here? :cowboy_hat_face:

Reply:
This is now formatted (copied from above)